Add `exports.types` to the package.json of `react-instantsearch`

### 🐛 Current behavior

Hi. We were trying out the [new beta native TS compiler (`@typescript/native-preview`)](https://www.npmjs.com/package/@typescript/native-preview), and the only error we are getting is to do with `react-instantsearch`:

```
error TS7016: Could not find a declaration file for module 'react-instantsearch'. '/Users/marcel/Code/owid-grapher/node_modules/react-instantsearch/dist/cjs/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/react-instantsearch` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-instantsearch';`

2 import { Snippet } from "react-instantsearch"
```

I have confirmed that this simple change to `react-instantsearch`'s package.json makes the issue go away:
```diff
@@ -8,6 +8,7 @@
   "type": "module",
   "exports": {
     ".": {
+      "types": "./dist/es/index.d.ts",
       "import": "./dist/es/index.js",
       "require": "./dist/cjs/index.js"
     }
```

Afaict, this shouldn't negatively affect anything else.

### 🔍 Steps to reproduce

1. Create a package that imports `react-instantsearch`
2. Run `npx @typescript/native-preview -b` on it

### Live reproduction

--

### 💭 Expected behavior

--

### Package version

react-instantsearch 4.20.2

### Operating system

_No response_

### Browser

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/packages/react-instantsearch-core/package.json
+++ b/packages/react-instantsearch-core/package.json
@@ -8,6 +8,7 @@
   "type": "module",
   "exports": {
     ".": {
+      "types": "./dist/es/index.d.ts",
       "import": "./dist/es/index.js",
       "require": "./dist/cjs/index.js"
     }

--- a/packages/react-instantsearch/package.json
+++ b/packages/react-instantsearch/package.json
@@ -8,6 +8,7 @@
   "type": "module",
   "exports": {
     ".": {
+      "types": "./dist/es/index.d.ts",
       "import": "./dist/es/index.js",
       "require": "./dist/cjs/index.js"
     }

